text/off-chain-workers/indexing.md

```diff
@@ -0,0 +1,97 @@
+# Off-chain Indexing
+
+There are times when on-chain extrinsics need to pass data to the off-chain worker context with
+predictable write behavior. We can surely pass this piece of data via on-chain storage, but this is
+costly and it will make the data propagate among the blockchain network. If this is not a piece of
+information that need to have consensus upon the whole network, another way is to save this data
```

Comment:
```suggestion
information that need to have consensus upon the whole network, another way is to save this data
```

indexing is a piece of information we've reach consensus about, it's just not polluting the on-chain state.